### PR TITLE
Mismatch improvements

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@alexander-aleksandrov, please review this PR. 

After we finish with it (and after you finish with current work), I want you to address TODOs in new Mismatch classes.

File: client/src/main/java/org/spine3/change/MessageMismatch.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+     * @return new {@code ValueMismatch} instance
+     */
+    public static ValueMismatch unexpectedValue(Message expected, Message actual, Message newValue, int version) {
+        checkNotNullOrEqual(expected, actual);
+        checkNotNull(newValue);
+
+        return of(expected, actual, newValue, version);
+    }
+
+    private static ValueMismatch of(Message expected, Message actual, Message newValue, int version) {
```
Comment (alexander-aleksandrov) [inline client/src/main/java/org/spine3/change/MessageMismatch.java]: There is no javadoc for this method.

File: client/src/main/java/org/spine3/change/StringMismatch.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+     * @return new ValueMismatch instance
+     */
+    public static ValueMismatch unexpectedValue(String expected, String actual, String newValue, int version) {
+        checkNotNullOrEqual(expected, actual);
+        checkNotNull(newValue);
+
+        return of(expected, actual, newValue, version);
+    }
+
+    private static ValueMismatch of(String expected, String actual, String newValue, int version) {
```
Comment (alexander-aleksandrov) [inline client/src/main/java/org/spine3/change/StringMismatch.java]: There is no javadoc for this method also.

**Comment by alexander-yevsyukov** [comment]
@alexander-aleksandrov PTAL

**Pull request merged**